Fix for #11 and #12

giuliobenetti opened the pull request:
Fix 0014 fixes #12
Fix 0015 fixes #11

giuliobenetti committed:
Fix 0014-mali-Make-devfreq-optional.patch

0014-mali-Make-devfreq-optional.patch can't be applied

This is due to wrong path specified inside the patch
that doesn't take into account mali version to be patched

Removed r6p2 from every path inside patch

Signed-off-by: Giulio Benetti <giulio.benetti@micronovasrl.com>

giuliobenetti commented:
@mripard 
Sorry but I really can't make it work last patch Reverse.
It gives me Unreversed patch detected!
It seems a dependency problem.
Unapply patches using same order than applying can bring to this.
Is there a way to make patch aware of dependency?

giuliobenetti committed:
Add unapply_patches function in build.sh

Build.sh doesn't correctly unapply patches

Patches can subsequently depend one on another.
Reversing patches in the same order as applied(top to bottom),
can generate problems.

Add unapply_patches function in build.sh
to start reverse patches from bottom to top.

Signed-off-by: Giulio Benetti <giulio.benetti@micronovasrl.com>

giuliobenetti commented:
@mripard added unapply_patches() for reversing patches from bottom to top.
Now everything seems to be ok.

mripard commented on patches/0015-mali-support-building-before-4.14.patch:
```diff
@@ -0,0 +1,42 @@
++	__setup_timer(timer, (TIMER_FUNC_TYPE)callback,
++		      (TIMER_DATA_TYPE)timer, flags);
++}
++#endif
```
That one shouldn't be part of a different patch, but fix the previous one.

mripard commented on patches/0013-mali-support-building-against-4.15.patch:
```diff
@@ -108,7 +108,7 @@ Signed-off-by: John Stultz <john.stultz@linaro.org>
 -	tim->timer.function = (timer_timeout_function_t)callback;
-+	timer_setup(&tim->timer, callback, 0);
++	__setup_timer(&tim->timer, callback, &tim->timer, 0);
  }
```
And using your previous patch, you dont need that one anymore

giuliobenetti committed:
Fix 0013-mali-support-building-against-4.15.patch

In 0013-mali-support-building-against-4.15.patch
has been used timer_setup api.
This api has been introduced only in 4.14.0 version,
so compiling with version < 4.14.0 fails.

Added function locally to mali_osk_timers.c compiled
if version < 4.14.0

Signed-off-by: Giulio Benetti <giulio.benetti@micronovasrl.com>

giuliobenetti commented:
@mripard 
Fix 0014 fixes #12
Fix 0013 fixes #11
Added also unapply_patches to reverse patches order to reverse patches

mripard commented on patches/0013-mali-support-building-against-4.15.patch:
```diff
@@ -1,17 +1,22 @@
+Date: Tue, 5 Dec 2017 22:32:31 +0100
+Subject: [PATCH 1/1] mali support building against 4.15-rc
+
+Used new timer api provided by <linux/timer.h>
```
You're not the primary author of this patch. You shouldn't change that part.

mripard commented on patches/0013-mali-support-building-against-4.15.patch:
```diff
@@ -1,17 +1,22 @@
+ 5 files changed, 24 insertions(+), 12 deletions(-)
 
+diff --git a/src/devicedrv/mali/common/mali_control_timer.c b/src/devicedrv/mali/common/mali_control_timer.c
+index fc6ceb4..57f3c3a 100755
```
Or that one.

mripard commented on patches/0013-mali-support-building-against-4.15.patch:
```diff
@@ -23,9 +28,11 @@ Signed-off-by: John Stultz <john.stultz@linaro.org>
 --- a/src/devicedrv/mali/common/mali_group.c
 +++ b/src/devicedrv/mali/common/mali_group.c
-@@ -44,7 +44,7 @@ int mali_max_job_runtime = MALI_MAX_JOB_
+@@ -44,7 +44,7 @@ int mali_max_job_runtime = MALI_MAX_JOB_RUNTIME_DEFAULT;
```
I guess you get the idea... :)

mripard commented on patches/0013-mali-support-building-against-4.15.patch:
```diff
@@ -1,17 +1,22 @@
+ src/devicedrv/mali/common/mali_osk_types.h      |  8 +++++++-
+ src/devicedrv/mali/linux/mali_memory_os_alloc.c |  2 +-
+ src/devicedrv/mali/linux/mali_osk_timers.c      | 16 ++++++++++------
+ 5 files changed, 24 insertions(+), 12 deletions(-)
```
Or that part.

giuliobenetti replied:
@mripard This depends on the fact I've added timer_setup function in case kernel_version < 4.14

giuliobenetti commented:
@mripard now everything should be ok!
Let me know, thank you
